adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala

```diff
@@ -156,6 +158,16 @@ class AlignmentRecordRDDFunctions(rdd: RDD[AlignmentRecord])
           conf
         )
     }
+    if (asRegularFile) {
+      log.info("make regular BAM/SAM file (not hadoop)")
+      val conf = new Configuration()
+      val fs = FileSystem.get(conf)
+      val ouputParentDir = filePath.substring(0, filePath.lastIndexOf("/") + 1)
+      val tmpPath = ouputParentDir + "tmp" + System.currentTimeMillis().toString
+      fs.rename(new Path(filePath + "/part-r-00000"), new Path(tmpPath))
```

Comment:
I thought I saw an instance where the format of the `part*` filename has varied across Hadoop versions; anyone know if that's a concern?